atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/atomic/TimestampExtractingAtomicTable.java

```diff
@@ -60,10 +64,19 @@ public void updateMultiple(Map<Long, Long> keyValues) throws KeyAlreadyExistsExc
     public ListenableFuture<Map<Long, Long>> get(Iterable<Long> keys) {
         return Futures.transform(
                 delegate.get(keys),
-                statuses -> KeyedStream.stream(statuses)
-                        .map(TransactionStatusUtils::maybeGetCommitTs)
-                        .flatMap(Optional::stream)
-                        .collectToMap(),
+                statuses -> {
+                    Set<TransactionStatus> unknowns = statuses.values()
+                            .stream()
+                            .filter(status -> status.equals(TransactionStatuses.unknown()))
+                            .collect(Collectors.toSet());
+                    Preconditions.checkState(unknowns.isEmpty(), "There has been a mistake in the wiring as "
+                            + "transactions that do not support transaction table sweep should not be seeing "
+                            + "`unknown` transaction status.");
```

Comment:
nit: don't need to collect to a set:

```suggestion
                    if (statuses.values().stream().anyMatch(TransactionStatuses.unknown()::equals)) {
                        throw new SafeIllegalStateException(
                                "There has been a mistake in the wiring as transactions that do not support "
                                        + "transaction table sweep should not be seeing `unknown` transaction status.",
                                SafeArg.of("statuses", statuses));
                    }
```